test-data/unit/fine-grained.test

```diff
@@ -3458,3 +3458,121 @@ a.py:1: error: "int" not callable
 [delete nonexistent_stub.pyi.2]
 [out]
 ==
+
+[case testDunderNewUpdatedMod]
+import a
+[file a.py]
+import b
+
+b.C(int())
+[file b.py]
+class C:
+    def __new__(cls, x: int) -> C:
+        pass
+[file b.py.2]
+class C:
+    def __new__(cls, x: str) -> C:
+        pass
+[out]
+==
+a.py:3: error: Argument 1 to "C" has incompatible type "int"; expected "str"
+
+[case testDunderNewUpdatedFunc]
```

Comment:
Not sure if we need both this and the above test case (`testMetaclassAttributesDirect`) since they are pretty similar.